osimpipeline/vital_tasks.py

```diff
@@ -986,109 +1162,342 @@ def __init__(self, trial, mrs_setup_task, **kwargs):
+        with util.working_directory(self.path):
             # On Mac, CmdAction was causing MATLAB ipopt with GPOPS output to
             # not display properly.
 
-            status = os.system('matlab -nodisplay -nodesktop -nosplash '
-                    '%s -logfile matlab_log.txt -r "try, '
+            status = os.system('matlab %s -logfile matlab_log.txt -wait -r "try, '
                     "run('%s'); disp('SUCCESS'); "
                     'catch ME; disp(getReport(ME)); exit(2), end, exit(0);"\n'
                     % ('-automation' if os.name == 'nt' else '',
-                       self.results_setup_fpath.replace('\\','/'))
+                        self.results_setup_fpath)
                     )
             if status != 0:
                 raise Exception('Non-zero exit status.')
 
+            # Wait until output mat file exists to finish the action
+            while True:
+                time.sleep(3.0)
+
+                mat_exists = os.path.isfile(self.results_output_fpath)
+                if mat_exists:
+                    break
+
+    def delete_muscle_analysis_results(self):
+        if os.path.exists(os.path.join(self.path, 'results')):
+            import shutil
+            shutil.rmtree(os.path.join(self.path, 'results'))
+
 class TaskMRSDeGrootePost(task.PostTask):
     REGISTRY = []
     def __init__(self, trial, mrs_setup_task, **kwargs):
         super(TaskMRSDeGrootePost, self).__init__(mrs_setup_task, trial, 
             **kwargs)
         self.doc = 'Postprocess DeGroote Muscle Redundancy Solver in MATLAB.'
         self.id = mrs_setup_task.tricycle.id
+        self.results_output_fpath = mrs_setup_task.results_output_fpath
+
+        # Plot muscle excitations, activations, and reserve activations
+        self.add_action([self.results_output_fpath],
+                        [os.path.join(self.path, 'muscle_activity.pdf'),
+                        os.path.join(self.path, 'reserve_activity.pdf')],
+                        self.plot_activations
+                        )
+
+        # Plot joint moment breakdown
+        self.add_action([self.results_output_fpath],
+                        [os.path.join(self.path, 'joint_moment_breakdown.pdf'),
+                        os.path.join(self.path, '%s_%s_mrs_moments.csv' %
+                        (self.study.name, mrs_setup_task.tricycle.id))],
+                        self.plot_joint_moment_breakdown
+                        )
+
+    def plot_activations(self, file_dep, target):
+
+        # Load mat file fields
+        muscle_names = util.hdf2list(file_dep[0], 'MuscleNames',isString=True)
+        exc = util.hdf2pandas(file_dep[0], 'MExcitation', labels=muscle_names)
+        act = util.hdf2pandas(file_dep[0], 'MActivation', labels=muscle_names)
+        dof_names = util.hdf2list(file_dep[0], 'DatStore/DOFNames', 
+            isString=True)
+        reserves = util.hdf2pandas(file_dep[0],'RActivation', labels=dof_names)
```

Comment:
Neat! Much better than using scipy's mat reader. Do you have a comment with your hdf functions mentioning that they can be used to read mat files. You said a minimum version of MATLAB is required; it might be good to document that.

Reply:
Not a minimum version of MATLAB, but .mat files need to be v7.3 to be HDF compatible. I'll include the note in the HDF functions.